src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheckTest.java

```diff
@@ -22,10 +22,14 @@
 import static com.google.common.truth.Truth.assertWithMessage;
 import static com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck.MSG_JAVADOC_MISSING;
 
+import java.io.File;
+
 import org.junit.jupiter.api.Test;
 
 import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtilTest;
 import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
 
 public class MissingJavadocMethodCheckTest extends AbstractModuleTestSupport {
```

Comment:
@relentless-pursuit looks like we missed migration of some tests, this is why pitest is failing:
https://github.com/checkstyle/checkstyle/blob/a168d97509e57bf78290f8fd790daf7f43c73030/src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java#L425-L446.

Please do:
1. Migrate all MissingJavadocMethod tests in CheckUtilTest to this class.
2. Migrate all inputs to corresponding package.

We should see a removal of test inputs from CheckUtil input package and addition of new inputs in MissingJavadocMethod package.

Reply:
done. i moved `InputCheckUtil9.java` and renamed it to match the convention. however, `InputCheckUtil1.java` is still being used by another method in `CheckUtilTest.java`. So, I created another file rather moving it. 

On a side note, what does it take to become one of the maintainers of checkstyle?

Reply:
Any idea on how to tackle the mutations because I assume i migrated all CheckUtilTests related to MissingJavadocMethod tests?
New surviving mutation(s) found:

```
Source File: "MissingJavadocMethodCheck.java"
Class: "com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck"
Method: "isSetterMethod"
Line Contents: "if (matchesSetterFormat &amp;&amp; singleParam) {"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF"
Description: "removed conditional - replaced equality check with true"
Line Number: 419
```

Reply:
>On a side note, what does it take to become one of the maintainers of checkstyle?

Just keep sending PRs with fixes for tickets.
While you are contributing you are maintainer. Your work will be visible at GitHub insights and release notes.

Reply:
Line in question is
https://github.com/checkstyle/checkstyle/pull/13607/files#diff-b4400c927ed8fca06e707dc1bc59acd1c05dba7eaaccfce7fb77956965760bcfR419

Pitest claims that changing this expression to "true" makes code to work the same ( non of test failing).
We need to find Input file content that can fail a test if such mutation is present.

Reply:
> matchesSetterFormat &amp;&amp; singleParam

Be careful which conditional it is changing. It could be changing `matchesSetterFormat` or `singleParam` and not the whole expression. You should confirm first before it turns out you are going down the wrong path.

Reply:
@romani, @rnveach thank you for such kind words and insights. 

i have been trying to write test cases to kill the mutation. 

for the recent push, i addded new assertions and they are correctly detecting mutations in my local, as they fail when the mutation is present but pass when the original condition is used. 

The first assertion which i ommitted (metioned below), passes in both the mutated and non-mutated cases. After omitting it, and running pitest locally, i didn't see any surviving mutations in suppression file. 
```
 assertWithMessage("Invalid result: AST provided is setter method")
                .that(MissingJavadocMethodCheck.isSetterMethod(setterMethod))
                .isTrue(); 
```

However, the above lines ensured to check a method which is a proper setter body. So, i have tried and spent a lot of time to tweak input file to kill mutation. Shall I proceed in some other direction? Or, suppressing the mutation can be a rationale choice?

Reply:
>suppressing the mutation can be a rationale choice?

We do now a lot of efforts to cleanup all suppressions. So we do not allow extension of them.

Reply:
@relentless-pursuit to be clear: we should have no need to add more tests or pitest suppressions. We previously had no surviving pitest mutations in this code. If we migrate all tests and test inputs, we should have none now. We are still missing tests and test cases in the migration effort. All test methods and test inputs should be moved to the correct path, verbatim.